vis-lua.c

```diff
@@ -1367,6 +1371,61 @@ static int redraw(lua_State *L) {
 	vis_redraw(vis);
 	return 0;
 }
+/***
+ * Closes a stream returned by @{Vis:communicate}.
+ *
+ * @function close
+ * @tparam io.file inputfd the stream to be closed
+ * @treturn bool identical to @{io.close}
+ */
+static int close_subprocess(lua_State *L) {
+	luaL_Stream *file = luaL_checkudata(L, -1, "FILE*");
+	if (file) {
+		lua_pop(L, 1);
```

Comment:
I'm sorry for  making you do this again but the `lua_pop()` is actually
incorrect. You just need:

```
if (!file)
	return luaL_argerror(...)
```

After reading `obj_ref_get()` in more detail I noticed it was modifying
the lua stack. This wasn't documented where it should be. It won't
matter to you but I am going to either add a comment or change it to
not do that so that I don't fall into that trap again.

Reply:
I looked to the Lua source code and now I do not even sure that remaining condition is necessary. The `luaL_checkudata` function never returns `NULL` according to [its source code](https://github.com/lua/lua/blob/v5.4.0/lauxlib.c#L345). 

It calls `luaL_testudata` which does return `NULL`, but after that it calls [a macro](https://github.com/lua/lua/blob/v5.4.0/lauxlib.h#L135) `luaL_argexpected` which will call the `luaL_typeerror` [function](https://github.com/lua/lua/blob/v5.4.0/lauxlib.c#L193) and eventually `luaL_argerror` with proper error message when the return value is `NULL`.

Reply:
Yep you are right. Sorry for the wild goose chase. I'm also pretty new
to the Lua-C API.

I do wish the documentation was a little more specific. The only indication
of this is [here][lua]:

> ... luaL_check* or luaL_opt*. All of these functions throw an error if the check is not satisfied.

Which doesn't really tell you that the function doesn't return in that case.

[lua]: https://www.lua.org/manual/5.1/manual.html#4